File: js/chunk_array.js
//see also https://www.codewars.com/kata/5298961d9ce954d77b0003a6/solutions/javascript
function chunk(arr, step){
  var tmp=arr.slice()
  var res=[]
  if (step<0){
    tmp.reverse();
    step=Math.abs(step);
  }
  for (let i=0;i<tmp.length;i += step)
    {
      res.push(tmp[i]);
    }
  return res;
}

File: cpp/chunk_array.cpp
#include <vector>

using namespace std;

static std::vector<int> chunk(std::vector<int> arr, int step) {
  std::vector<int> a, o;
  a.insert(a.end(),arr.begin(),arr.end());
  if (step<0){
    reverse(a.begin(),a.end());
    step=-step;
  }
  for (unsigned long i=0;i<a.size();i+=step){
    o.push_back(a[i]);
  }
  return o;
}
